vendor/update-odoc-parser.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+version=ebfd3b9489e44187da2c67d79a32b6fc1e92bda4
+
+set -e -o pipefail
+
+TMP="$(mktemp -d)"
+trap "rm -rf $TMP" EXIT
+
+rm -rf odoc-parser
+mkdir -p odoc-parser/src
+
+(
+    cd $TMP
+    git clone https://github.com/ocaml-doc/odoc-parser.git
+    cd odoc-parser
+    git checkout $version
+)
+
+SRC=$TMP/odoc-parser
+
+cp -v $SRC/src/*.{ml,mli,mll} odoc-parser/src
+cp -v $SRC/LICENSE.md odoc-parser/
+
+git checkout odoc-parser/src/dune
+git add -A .
```

Comment:
I'm not sure, what is this line for?

Reply:
Which, the git add? I should mention that this script was modelled closely on (copied from!) dune's equilvalent scripts, e.g. https://github.com/ocaml/dune/blob/main/vendor/update-cmdliner.sh

Reply:
Yes, I was wondering why automatically adding those files to the git index... (I'm OK with it, I just wanted to know if there was a specific reason you knew for this)